### PR TITLE
Modify dd_timeout and the way getting qtree num_queues info…

### PR DESCRIPTION
**Pull request by pezhang**
Fix dd timeout issue and update find virtqueue method 
1. increase dd_timeout
2. get queue number from virtio-scsi-pci device
3. change dd jobs from sequential to parallel

Signed-off-by: Pei Zhang <pezhang@redhat.com>

### CODE REVIEW

**Comment by xutian** [comment]
ID:  1302506

**Comment by xutian** [comment]
@pezhang   autopep8 check failed, please fix and force push it again.

**Comment by xutian** [comment]
Please update your commit message

**Comment by pezhang** [comment]
@xutian , I changed the commit message.

File: qemu/tests/virtio_scsi_mq.py [hunk 1]
```diff
@@ -177,6 +176,9 @@ def proc_interrupts_results(results):
         raise error.TestError("Disks are not all show up in system. Output "
                               "from the check command: %s" % output)
     fill_cmd = fill_cmd.rstrip("&&")
+    # As Bug 1177332 exists, mq is not supported completely.
+    # So don't considering performance currently, dd_timeout is longer.
+    dd_timeout = dd_timeout * (images_num * 2)
```
Comment (xutian) [inline qemu/tests/virtio_scsi_mq.py]: any difference between   a * b *c and a * (b * c) ? it looks there no need to add  '()'.

Thanks,
Xu

File: qemu/tests/virtio_scsi_mq.py [hunk 2]
```diff
@@ -122,21 +121,21 @@ def proc_interrupts_results(results):
...
-                                                     pcic.qtree["num_queues"])
-                            raise error.TestFail(error_msg)
-                    if pcic.qtree["class_name"] == "SCSI controller":
-                        scsi_bus_addr = pcic.qtree['addr']
-                        break
-
+    qtree_num_queues_full = ""
+    qtree_num_queues = ""
+    for node in qtree.get_nodes():
+        if (isinstance(node, qemu_qtree.QtreeDev)) and (
```
Comment (xutian) [inline qemu/tests/virtio_scsi_mq.py]: Does this fix works with qemu in rhel6 host?

Thanks,
Xu

**Comment by pezhang** [comment]
@xutian , this case is only for rhel7 host. And I have removed "()" you mentioned to make the code clear.  Thanks a lot.

File: qemu/tests/virtio_scsi_mq.py [hunk 3]
```diff
@@ -122,21 +121,21 @@ def proc_interrupts_results(results):
...
+                node.qtree['type'] == "virtio-scsi-device"):
+            qtree_num_queues_full = node.qtree["num_queues"]
+            qtree_num_queues = qtree_num_queues_full.split("(")[0].strip()
```
Comment (xutian) [inline qemu/tests/virtio_scsi_mq.py]: can we git it via 're.search' ?

Comment (pezhang) [inline qemu/tests/virtio_scsi_mq.py]: I can have a try.

Comment (pezhang) [inline qemu/tests/virtio_scsi_mq.py]: @xutian , I have updated the code using 're.search'

File: qemu/tests/virtio_scsi_mq.py [hunk 4]
```diff
@@ -122,21 +122,23 @@ def proc_interrupts_results(results):
...
+    qtree_num_queues_full = ""
+    qtree_num_queues = ""
+    for node in qtree.get_nodes():
+        if (isinstance(node, qemu_qtree.QtreeDev)) and (
+                node.qtree['type'] == "virtio-scsi-device"):
+            qtree_num_queues_full = node.qtree["num_queues"]
+            qtree_num_queues = re.search(
+                "[0-9]+",
+                qtree_num_queues_full).group()
+        elif (isinstance(node, qemu_qtree.QtreeDev)) and (node.qtree['type'] == "virtio-scsi-pci"):
```
Comment (xutian) [inline qemu/tests/virtio_scsi_mq.py]: No '()' need here.

Comment (pezhang) [inline qemu/tests/virtio_scsi_mq.py]: @xutian , Have removed it. And thanks.

**Comment by xutian** [comment]
looks ok to me

File: qemu/tests/virtio_scsi_mq.py [hunk 5]
```diff
@@ -171,14 +173,21 @@ def proc_interrupts_results(results):
...
     if count != images_num:
         raise error.TestError("Disks are not all show up in system. Output "
                               "from the check command: %s" % output)
-    fill_cmd = fill_cmd.rstrip("&&")
+    # As Bug 1177332 exists, mq is not supported completely.
+    # So don't considering performance currently, dd_timeout is longer.
+    dd_timeout = dd_timeout * images_num * 2
     session.cmd(fill_cmd, timeout=dd_timeout)
 
+    dd_thread_num = count
```
Comment (xutian) [inline qemu/tests/virtio_scsi_mq.py]: why need this step?  do you want to  wait for dd threads terminate ?

Comment (pezhang) [inline qemu/tests/virtio_scsi_mq.py]: yes, it's for waiting for all dd finish off.

**Comment by XueqiangWei** [comment]
LGTM,
Acked by Xueqiang Wei<xuwei@redhat.com>

**Comment by xutian** [comment]
looks fine, merged. thanks @pezhang and @XueqiangWei

**Pull request merged**